outputs.tf

```diff
@@ -71,6 +49,41 @@ output "security_group_policies" {
   })
 }
 
+output "kubernetes_secrets" {
+  sensitive   = true
+  description = "Kubernetes secrets required for the Spacelift services."
+  value = templatefile("${path.module}/kubernetes-secrets.tftpl", {
+    namespace                                      = var.k8s_namespace
+    AWS_ACCOUNT_ID                                 = data.aws_caller_identity.current.account_id
+    AWS_REGION                                     = var.aws_region
+    SERVER_DOMAIN                                  = var.server_domain
+    MQTT_BROKER_DOMAIN                             = coalesce(var.mqtt_broker_domain, "spacelift-mqtt.${var.k8s_namespace}.svc.cluster.local")
+    ENCRYPTION_TYPE                                = "kms"
+    ENCRYPTION_KMS_ENCRYPTION_KEY_ID               = module.spacelift.kms_encryption_key_arn
+    ENCRYPTION_KMS_SIGNING_KEY_ID                  = module.spacelift.kms_signing_key_arn
+    OBJECT_STORAGE_BUCKET_DELIVERIES               = module.spacelift.deliveries_bucket_name
+    OBJECT_STORAGE_BUCKET_LARGE_QUEUE_MESSAGES     = module.spacelift.large_queue_messages_bucket_name
+    OBJECT_STORAGE_BUCKET_MODULES                  = module.spacelift.modules_bucket_name
+    OBJECT_STORAGE_BUCKET_POLICY_INPUTS            = module.spacelift.policy_inputs_bucket_name
+    OBJECT_STORAGE_BUCKET_RUN_LOGS                 = module.spacelift.run_logs_bucket_name
+    OBJECT_STORAGE_BUCKET_STATES                   = module.spacelift.states_bucket_name
+    OBJECT_STORAGE_BUCKET_USER_UPLOADED_WORKSPACES = module.spacelift.user_uploaded_workspaces_bucket_name
+    OBJECT_STORAGE_BUCKET_WORKSPACE                = module.spacelift.workspace_bucket_name
+    OBJECT_STORAGE_BUCKET_METADATA                 = module.spacelift.metadata_bucket_name
+    OBJECT_STORAGE_BUCKET_UPLOADS                  = module.spacelift.uploads_bucket_name
+    OBJECT_STORAGE_BUCKET_UPLOADS_URL              = module.spacelift.uploads_bucket_url
+    DATABASE_URL                                   = module.spacelift.database_url
+    DATABASE_READ_ONLY_URL                         = module.spacelift.database_read_only_url
+    LICENSE_TOKEN                                  = var.license_token != null ? var.license_token : ""
+    SPACELIFT_PUBLIC_API                           = var.spacelift_public_api != null ? var.spacelift_public_api : ""
+    WEBHOOKS_ENDPOINT                              = "https://${var.server_domain}/webhooks"
+    LAUNCHER_IMAGE                                 = module.spacelift.ecr_launcher_repository_url
+    SPACELIFT_VERSION                              = var.spacelift_version != null ? var.spacelift_version : ""
```

Comment:
default value is an empty string, so this probably won't be null

Reply:
can't this be just?
```suggestion
    SPACELIFT_VERSION                              = var.spacelift_version
```

Reply:
Yeah - but someone can still set it to null, so I figured it was safer to just check.

Reply:
The problem is that if any of these variables are null, the template function errors out, so I just wanted to avoid that.